### PR TITLE
rename metrics_server to listen_address

### PR DESCRIPTION
**Pull request by tipmg**
According to official documentation - https://docs.gitlab.com/runner/monitoring/README.html#configuration-of-the-metrics-http-server

### CODE REVIEW

**Comment by LongLiveCHIEF** [comment]
@tipmg can you fix the failing tests? Looks like it is just some linting issues.  Other than that, this one LGTM :+1:

File: manifests/init.pp [hunk 1]
```diff
@@ -92,7 +92,7 @@
         }
       }
       default: {
-        fail ("gitlab_ci_runner::manage_repo parameter for ${::osfamily} is not supported.")
+        fail("gitlab_ci_runner::manage_repo parameter for ${::osfamily} is not supported.")
```
Comment (Dan33l) [inline manifests/init.pp]: This change does not look related to this PR.
If this line is changed we needs to call the fact via $facts and remove leading `::`.
IMO this should be in an other PR.

File: manifests/init.pp [hunk 2]
```diff
@@ -111,13 +111,14 @@
...
-  if $metrics_server {
-    file_line { 'gitlab-runner-metrics-server':
-      path    => '/etc/gitlab-runner/config.toml',
-      line    => "metrics_server = \"${metrics_server}\"",
-      match   => '^metrics_server = .+',
-      require => Package[$package_name],
-      notify  => Exec['gitlab-runner-restart'],
+  if $listen_address {
+    file_line { 'gitlab-runner-listen-address':
+      path              => '/etc/gitlab-runner/config.toml',
```
[the remark this answers is not in the document]
Comment (tipmg) [reply manifests/init.pp]: Hello @Dan33l 
I fixed it.

File: manifests/init.pp (hunk 1 above)
Comment (tipmg) [reply manifests/init.pp]: Yes, you are right. Thanks. I fixed it too.

Comment (tipmg) [reply manifests/init.pp]: It is interesting case. Test will be fail. I returned ${::osfamily}
I recommend to use $facts['os']['family']
It is decision by author.

**Comment by tipmg** [comment]
Hello, @LongLiveCHIEF I fixed tests.

**Comment by tipmg** [comment]
Hello, @Dan33l Please review changes.

File: manifests/init.pp (hunk 1 above)
Comment (Dan33l) [reply manifests/init.pp]: It is strange when i run tests locally with `fail("gitlab_ci_runner::manage_repo parameter for ${facts['os']['family']} is not supported.")` the command ` bundle exec rake test` does not fail.

What was the problem you encountered ?

**Comment by bastelfreak** [comment]
Also this looks like a breaking-change. I'm not sure if that move is required.

**Comment by LongLiveCHIEF** [comment]
We deal with this a bit in the `puppet-gitlab` module also, and so we came up with the following strategy, taken from [puppet-gitlab/README.md/#description](https://github.com/voxpupuli/puppet-gitlab#module-description):

> This module is designed to support the most recent versions of the gitlab-omnibus package (both ce and ee). Gitlab will support and release patches for the last 3 releases. This module can typically support the most recent major version, as well as the previous major version, but is currently only tested in the gitlab-supported versions of the module.

One of the reasons I broke this out into its own module however, was the ability to make different decisions for the runners. Just because someone is using this module, does not mean they're using the `puppet-gitlab` module, and it is very likely someone could use a v10 runner with v11 of GitLab. (I've actually seen it _a lot_).

This opens up the discussion of what the support strategy should be for _this_ module. 

I will open a discussion issue for that discussion, but in the meantime, I don't think we can leave both, and make a note in the README that users should be aware and use the correct parameters for their version.

@tipmg can you ammend the PR to leave in the `metrics_server` param, while still adding the `listen_address`?

**Comment by alexjfisher** [comment]
> can you amend the PR to leave in the metrics_server param, while still adding the listen_address?
 +1

**Comment by dhollinger** [comment]
I think we should keep the `metrics_server` param in for now, but revisit it soon. After spending some time in the Gitlab docs, they seem to imply that (aside from critical security fixes and certain pay levels) they primarily only support latest stable, which is a monthly release.

While I don't think we should be that aggressive, we should discourage the use of major releases older than 18-24 months (or less). Each Gitlab major release is yearly, so that should be simple enough. Additionally, we should only support latest stable minor release, aside from critical bugfixes (that shouldn't be difficult since Gitlab only usually introduces breaking changes in major releases)

**Comment by baurmatt** [comment]
Any update on this? :)
